steps/kubernetes.py

```diff
@@ -0,0 +1,200 @@
+'''test methods for kubernetes'''
+
+import re
+import json
+import time
+from jinja2 import Environment, FileSystemLoader
+from behave import *
+
+# NB: Note that we aggregate across masters. This doesn't matter in reality
+# since our cluster test uses only one master. But if we ever change to using
+# multiple masters, things will be different.
+
+def kubectl_get(context, component):
+    r = context.remote_cmd('command',
+                           module_args='kubectl get %s -o json' % component)
+    if r:
+        items = []
+        for i in r:
+            data = json.loads(i['stdout'])
+            if 'items' in data:
+                items.extend(data['items'])
+        return items
+    else:
+        assert False, "unable to get %s" % component
+
+def kubectl_create(context, resource):
+    r = context.remote_cmd('command',
+                           module_args='kubectl create -f %s' % resource)
+    assert r, "could not create resource %s" % resource
+
+def kubectl_exec(context, pod_name, command):
+    # NB: do not be confused by the r[i]['rc'] value. The `kubectl exec` command
+    # will always return 0 even if the exec'ed command returns a non-zero. If
+    # r[i]['rc'] is nonzero, then it means that something went wrong with
+    # kubernetes itself (e.g. pod doesn't exist).
+    r = context.remote_cmd('command',
+        module_args='kubectl exec %s -- %s' % (pod_name, command))
```

Comment:
The indent will break PEP8 check in python. Maybe this is better:
    r = context.remote_cmd('command',
                                            module_args='kubectl exec %s -- %s' % (pod_name, 
                                                                                                              command))